pkg/backend/backend.go

```diff
@@ -83,21 +85,28 @@ func (b *backend) Run(ctx context.Context, stop <-chan struct{}) {
 		b.mu.Unlock()
 
 		if b.stopping.Load().(bool) {
-			break
-		}
-
-		ocbDidWork, err := b.ocb.try(ctx)
-		if err != nil {
-			b.baseLog.Error(err)
-		}
-
-		sbDidWork, err := b.sb.try(ctx)
-		if err != nil {
-			b.baseLog.Error(err)
-		}
-
-		if !(ocbDidWork || sbDidWork) {
-			<-t.C
+			if b.workers == 0 {
+				b.baseLog.Infof("all backend workers exited. Terminating backend mananger")
+				close(done)
+				break
+			} else {
+				b.baseLog.Infof("backend workers still running. Current count: %d", b.workers)
+				time.Sleep(30 * time.Second)
+			}
```

Comment:
move this out of the for loop to the bottom of the function and do something like this instead of the sleep:
```
for {
	b.mu.Lock()
	for atomic.LoadInt32(&b.workers) > 0 {
		b.cond.Wait()
		// possibly log the number of workers left here, but shouldn't we see that with the metrics?
	}
	b.mu.Unlock()
}
close(done)
```